src/SignalR/common/Shared/MessageBuffer.cs

```diff
@@ -89,10 +100,19 @@ private async Task RunTimer()
                     await _writeLock.WaitAsync().ConfigureAwait(false);
                     try
                     {
+                        // No need to try and send when we know the connection is down
+                        if (!_resend.Task.IsCompleted)
+                        {
+                            continue;
+                        }
+
                         _protocol.WriteMessage(_ackMessage, _connection.Transport.Output);
                         await _connection.Transport.Output.FlushAsync().ConfigureAwait(false);
                         _lastAckedId = sequenceId;
                     }
+                    // Ignore Pipe errors. The connection closed but might come back, if it doesn't come back MessageBuffer with be disposed
+                    // which disposes the timer.
+                    catch { }
```

Comment:
These are errors on flush right?

Reply:
Yes

Reply:
Did you observe this? A socket connection and ResponseBodyPipeWriter should never throw on flush unless it's cancelled.